nautobot/extras/jobs.py

```diff
@@ -104,25 +105,190 @@ class Meta:
 
+    # TODO(jathan): Could be interesting for custom stuff when the Job is
+    # enabled in the database and then therefore registered in Celery
+    @classmethod
+    def on_bound(cls, app):
+        """Called when the task is bound to an app.
+
+        Note:
+            This class method can be defined to do additional actions when
+            the task class is bound to an app.
+        """
+
+    # TODO(jathan): Could be interesting for showing the Job's class path as the
+    # shadow name vs. the Celery task_name?
+    def shadow_name(self, args, kwargs, options):
+        """Override for custom task name in worker logs/monitoring.
+
+        Example:
+            .. code-block:: python
+
+                from celery.utils.imports import qualname
+
+                def shadow_name(task, args, kwargs, options):
+                    return qualname(args[0])
+
+                @app.task(shadow_name=shadow_name, serializer='pickle')
+                def apply_function_async(fun, *args, **kwargs):
+                    return fun(*args, **kwargs)
+
+        Arguments:
+            args (Tuple): Task positional arguments.
+            kwargs (Dict): Task keyword arguments.
+            options (Dict): Task execution options.
+        """
+
+    def before_start(self, task_id, args, kwargs):
+        """Handler called before the task starts.
+
+        .. versionadded:: 5.2
+
+        Arguments:
+            task_id (str): Unique id of the task to execute.
+            args (Tuple): Original arguments for the task to execute.
+            kwargs (Dict): Original keyword arguments for the task to execute.
+
+        Returns:
+            None: The return value of this handler is ignored.
+        """
+        self.active_test = "initialization"
+
+        try:
+            self.job_result = self.get_job_result()
+        except TypeError as err:
+            raise RunJobTaskFailed(f"Unable to serialize data for job {task_id}") from err
+        except Exception as err:
+            raise RunJobTaskFailed(f"Unexpected failure in job {self.name}") from err
+
+        job_model = self.get_job_model()
+        if not job_model.enabled:
+            self.log_failure(
+                message=f"Job {job_model} is not enabled to be run!",
+                obj=job_model,
+            )
+
+        soft_time_limit = job_model.soft_time_limit or settings.CELERY_TASK_SOFT_TIME_LIMIT
+        time_limit = job_model.time_limit or settings.CELERY_TASK_TIME_LIMIT
+        if time_limit <= soft_time_limit:
+            self.log_warning(
+                f"The hard time limit of {time_limit} seconds is less than "
+                f"or equal to the soft time limit of {soft_time_limit} seconds. "
+                f"This job will fail silently after {time_limit} seconds.",
+            )
+
+        self.log_info("Running job")
+
+    def run(self, *args, **kwargs):  # pylint: disable=arguments-differ
```

Comment:
Hmm. What's pylint expect the signature to be?

Reply:
I *think* the problem is that pylint doesn't understand that the base class `celery.app.task.Task` is an abstract base class so it doesn't like that individual jobs are redefining the signature. I ended up disabling this check for the entire project temporarily because I couldn't find a way to make pylint happy. This was probably left in mistakenly while trying to figure out a generic fix because this run method never generated a pylint warning.